pkg/envoy/ads/stream.go

```diff
@@ -190,23 +193,62 @@ func respondToRequest(proxy *envoy.Proxy, discoveryRequest *xds_discovery.Discov
 	// than what we last sent. (Perhaps the control plane restarted.)
 	// In that case we want to make sure that we send new responses with
 	// VersionInfo incremented starting with the version which the proxy last had.
-	if ackVersion > proxy.GetLastSentVersion(typeURL) {
-		proxy.SetLastSentVersion(typeURL, ackVersion)
+	lastVersion = proxy.GetLastSentVersion(typeURL)
+	if requestVersion > lastVersion {
+		log.Debug().Msgf("Proxy SerialNumber=%s PodUID=%s: Higher version on request %s, req ver: %d - last ver: %d. Updating to match latest.",
+			proxy.GetCertificateSerialNumber(), proxy.GetPodUID(), typeURL.Short(), requestVersion, lastVersion)
+		proxy.SetLastSentVersion(typeURL, requestVersion)
+		return true
 	}
 
-	lastNonce := proxy.GetLastSentNonce(typeURL)
-	if lastNonce != "" && discoveryRequest.ResponseNonce == lastNonce {
-		log.Debug().Msgf("Proxy SerialNumber=%s PodUID=%s: Nothing changed for Envoy on Pod since Nonce=%s",
-			proxy.GetCertificateSerialNumber(), proxy.GetPodUID(), discoveryRequest.ResponseNonce)
+	// Compare Nonces
+	// As per protocol, we can ignore any request on the TypeURL stream that has not caught up with last sent nonce, if the
+	// nonce is non-empty.
+	// A newly connecting envoy might send a version and nonce from an old control plane, this is currently mitigated by the
+	// forced update @line 69
```

Comment:
Line number is going to get outdated pretty quick, might as well paste the line here.